components/locid/src/helpers.rs

```diff
@@ -458,37 +459,40 @@ macro_rules! impl_tinystr_subtag {
         /// Parsing errors don't have to be handled at runtime:
         /// ```
         /// assert_eq!(
-        #[doc = concat!("  icu_locid::", stringify!($macro_name), "!(", stringify!($good_example) ,"),")]
-        #[doc = concat!("  ", stringify!($good_example), ".parse::<icu_locid::", stringify!($($full_name)::+),">().unwrap()")]
+        #[doc = concat!("  icu_locid::", $(stringify!($path), "::",)+ stringify!($macro_name), "!(", stringify!($good_example) ,"),")]
+        #[doc = concat!("  ", stringify!($good_example), ".parse::<icu_locid::", $(stringify!($path), "::",)+ stringify!($name), ">().unwrap()")]
         /// );
         /// ```
         ///
         /// Invalid input is a compile failure:
         /// ```compile_fail,E0080
-        #[doc = concat!("icu_locid::", stringify!($macro_name), "!(", stringify!($bad_example) ,");")]
+        #[doc = concat!("icu_locid::", $(stringify!($path), "::",)+ stringify!($macro_name), "!(", stringify!($bad_example) ,");")]
         /// ```
         ///
-        #[doc = concat!("[`", stringify!($name), "`]: crate::", stringify!($($full_name)::+))]
+        #[doc = concat!("[`", stringify!($name), "`]: crate::", $(stringify!($path), "::",)+ stringify!($name))]
         #[macro_export]
-        macro_rules! $macro_name {
+        #[doc(hidden)]
+        macro_rules! $legacy_macro_name {
             ($string:literal) => {{
-                use $crate::$($full_name)::+;
+                use $crate::$($path ::)+ $name;
                 const R: $name =
                     match $name::try_from_bytes($string.as_bytes()) {
                         Ok(r) => r,
                         #[allow(clippy::panic)] // const context
-                        _ => panic!(concat!("Invalid ", stringify!($name), ": ", $string)),
+                        _ => panic!(concat!("Invalid ", $(stringify!($path), "::",)+ stringify!($name), ": ", $string)),
                     };
                 R
             }};
         }
+        #[doc(inline)]
+        pub use $legacy_macro_name as $macro_name;
```

Comment:
Praise: Seems so obvious in hindsight; and I guess it works in Rust 1.61 or else CI would fail

Reply:
This has probably been possible since 1.0.

Reply:
haven't we been using this trick already? I definitely recall mentioning it and I think baked_exporter uses it a bunch

Reply:
We thought this wasn't possible, see https://github.com/unicode-org/icu4x/issues/253#issuecomment-1133207454. I figure out how to do this for baked exporter, and now am applying it here.